docs/prover/cli/options.md

```diff
@@ -369,17 +369,60 @@ certoraRun Bank.sol --verify Bank:Bank.spec --optimistic_loop
 
-**When to use it?**  
-The default number of loop iterations we unroll is one. However, in many cases, bugs only occur when there are several iterations. Common scenarios include iteration over list elements. Two, or in some cases three, is usually the most you will ever need to uncover bugs.  
+**When to use it?**
+The default number of loop iterations we unroll is one. However, in many cases, bugs only occur when there are several iterations. Common scenarios include iteration over list elements. Two, or in some cases three, is usually the most you will ever need to uncover bugs.
 
 **Example**
 
 ```
 certoraRun Bank.sol --verify Bank:Bank.spec --loop_iter 2
 ```
 
+Options regarding summarization
+-------------------------------
+
+(--optimistic_summary_recursion)=
+### `--optimistic_summary_recursion`
+
+**What does it do?**
+In case there's a call to some Solidity function within a summary, we may end up with recursive calls to this summary. For example, if in the summary of `foo` we call the Solidity function `bar`, and `bar`'s Solidity code contains a call to `foo`, we'll summarize `foo` again, which will lead to another call to `bar` etc. In this case if this flag is set to `false` we may get an assertion failure with a message along the lines of
+```
+Recursion limit (...) for calls to ..., reached during compilation of summary ...
+```
+Alternatively, such recursion could happen with {ref}`dispatcher` summaries - if both contract A and B have a function `foo()`, and `A.foo()` contains a call to `someContractAddress.foo()`, then if `foo()` is summarized with {ref}`dispatcher`, and `B.foo()` is unresolved, the dispatching may choose `A.foo()` as one of the possible targets and we end up with recursion. In this case if this flag is set to `false` we may get an assertion failure message along the lines of
+```
+When summarizing a call with dispatcher, found we already have it in the stack: ... consider removing its dispatcher summary.
+```
+The default behavior in this case is to assert if the recursion limit is reached (the limit is controlled by the {ref}`--summary_recursion_limit` flag). Setting this flag to `true` will instead assume that the limit is never reached.
+
+**When to use it**
+There's recursion due to summaries calling Solidity functions, and this leads to an assertion failure due to this. In this case one can either make the limit larger (via {ref}`--summary_recursion_limit`) or set this flag to `true`.
+
+**Example**
+
+```
+certoraRun Bank.sol --verify Bank:Bank.spec --optimistic_summary_recursion true
+```
+(--summary_recursion_limit)=
+### `--summary_recursion_limit`
+
+**What does it do?**
+In case there's a call to some Solidity function within a summary, we may end up with recursive calls to this summary. For example, if in the summary of `foo` we call the Solidity function `bar`, and `bar`'s Solidity code contains a call to `foo`, we'll summarize `foo` again, which will lead to another call to `bar` etc.
```

Comment:
I think instead of duplicating the text here, it might be better to just reference it.  E.g. ``Summaries can cause recursion (see {ref}`--optimistic_summary_recurion`).  This option sets the summary recursion level, which is the number of recursive calls that the Prover will consider.

If a counterexample causes a function to be called recursively more than the summary recursion limit, it will report an assertion failure (unless {ref}`--optimistic_summary_recursion` is set, in which case the counterexample will be ignored).